packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModuleConstantsHelper.java

```diff
@@ -50,6 +52,31 @@
         DIRECT_EVENTS_KEY, UIManagerModuleConstants.getDirectEventTypeConstants());
   }
 
+  private static void validateDirectEventNames(
+      String viewManagerName, Map<String, Object> directEvents) {
+    if (!ReactBuildConfig.DEBUG || directEvents == null) {
+      return;
+    }
+
+    for (String key : directEvents.keySet()) {
+      Object value = directEvents.get(key);
+      if (value != null && (value instanceof Map)) {
+        String regName = (String) ((Map) value).get("registrationName");
+        if (regName != null
+            && key.startsWith("top")
+            && regName.startsWith("on")
+            && !key.substring(3).equals(regName.substring(2))) {
```

Comment:
Just for curiosity;
Should this be simplified as well, what is being checked here?